schema.graphql

```diff
@@ -73,31 +79,63 @@ type VerseText {
   assessmentDetails: [AssessmentDetail]
 }
 
-enum AssessmentType {
+enum TextAnnotationType {
   SemanticSimilarity
   QAComprehensibility
   SubwordsPerSentence
+  VersePart
+  PartOfSpeech
+  NamedEntity
+}
+
+enum ReferenceAnnotationType {
+  SemanticDomain
+  KeyTerm
 }
 
 type Assessment {
   id: ID!
-  type: AssessmentType! @search(by: [exact])
+  type: TextAnnotationType! @search(by: [exact])
   bibleRevision: BibleRevision!
   reference: BibleRevision
   date: DateTime! @search
-  details: [AssessmentDetail!] @hasInverse(field: assessment)
+  details: [Annotation!] @hasInverse(field: assessment)
+}
+
+interface Annotation {
+  id: ID!
+  flag: Boolean @search
+  number: Float @search
+  tag: String @search
+  parent: [Annotation] @hasInverse(field: children)
+  children: [Annotation] @hasInverse(field: parent)
 }
 
-type AssessmentDetail {
+type TextAnnotation implements Annotation {
   id: ID!
-  assessment: Assessment!
-  assessmentMetrics: [AssessmentMetric!] @hasInverse(field: assessmentDetail)
-  verseTexts: [VerseText]
+  verseTexts: [VerseText!]
```

Comment:
Now that the Annotation includes a span, should this be tied to only a single `verseText` rather than multiple? Because otherwise, it will be unclear where the span is occurring (I think).

Reply:
The thought was that the verseTexts were sortable via the information in the reference chain on chapter/verse numbers and that if multiple verses the start position for the span would be in the first verse and the end position would be in the last verse.

Reply:
What if the annotation is of a semantic domain that occurs in two different places in the verse range. I guess that would correspond to two different annotations? 

I guess there is also the "range" issue. So even a semantic similarity annotation could correspond to multiple verses. Is that right? I'm not sure about the internal span within a verse range, but the score for a verse range makes sense to me.

Reply:
After thinking this through, I think I'm with you. Let's see how it plays out.